sherpa/tests/test_data_notebook.py

```diff
@@ -26,15 +27,28 @@
 from sherpa import data
 from sherpa import plot
 
+# This not the most elegant solution, but it makes sense to
+# have an independent check here and not rely on what is
+# done in the sherpa.plot.__init__ module, so that the tests
+# stay independent of that particular implementation.
+try:
+    from sherpa.plot.pylab_backend import PylabBackend
+    HAS_PYLAB = True
+except ModuleNotFoundError:
+    HAS_PYLAB = False
 
-def test_data1d(old_numpy_printing, override_plot_backend):
+
+def test_data1d(old_numpy_printing, all_plot_backends):
     d = data.Data1D("x x",
                     np.asarray([1, 3, 6]),
                     np.asarray([3, 4, 7]))
     r = d._repr_html_()
     assert r is not None
 
-    if plot.backend.name == 'pylab':
+    # If not HAS_PYLAB, then the symbol PylabBackend is not defined
+    # and test collection will fail. So, we short-circuit that
+    # such that PylabBackend will only be needed if HAS_PYLAB is True.
+    if HAS_PYLAB and isinstance(plot.backend, PylabBackend):
```

Comment:
This is used in enough places that it could be made a fixture (that returns True or False) that would be used something like

```
def test_foo(...., using_pylab):
...
    if using_pylab:
        blah
    else:
        other-blah
```

but I'm not going to push it for this PR

Reply:
We need to think about this if we add more backends that are derived from PylabBackend.  Right now, we only have one (pylab_area_backend) that's mostly used to demonstration. However, if we do add more options, ten we don't want to run all tests for all of them, but just some, so we'll have to take that into account.